vignettes/kaplan_meier.Rmd

```diff
@@ -8,25 +8,45 @@ vignette: >
   %\VignetteEncoding{UTF-8}
 ---
 
+# 1. Data
+
 ```{r, include=FALSE, warning=FALSE}
 knitr::opts_chunk$set(
   collapse = TRUE,
-  comment = "#>"
-)
+  comment = "#>")
 
 library(tidyverse)
 t0 = ISOdate(2020, 01, 01, tz="EST")
+t0_v2=as.Date(t0, format =  "%Y/%m/%d")
+# , format(Sys.time(), "%*Y/%m/%d"
 data_surv = ggsurvfit::df_colon %>% 
   mutate(
-    date_enrol = t0 + abs(rnorm(n(), 100, 10)), 
-    date_eos = date_enrol + time,
-    death_eos = status,
-    date_progression = date_enrol + time*runif(n(), 0.6, 0.8),
-    date_progression = if_else(runif(n())>0.2, NA, date_progression), 
+    date_first_visit = t0_v2 + abs(rnorm(n(), 500, 20)), 
+    date_of_last_visit = date_first_visit + time,
+    death = status,
+    date_progression = date_first_visit + time*runif(n(), 0.6, 0.8),
+    # date_progression = if_else(runif(n())>0.2, NA, date_progression), 
   ) %>% 
-  select(subijd=id, arm=rx, surg, 
-         date_enrol, date_progression, date_eos, death_eos)
-#data_surv est caché (include=FALSE), il suffit donc de présenter le dataset et ses colonnes (comme tu le ferais avec df_colon), puis d'expliquer comment utiliser pmin() pour créer time_os, event_os, time_pfs, event_pfs
+  select(subijd=id,arm=rx, surg, 
+         date_first_visit,time, date_progression, date_of_last_visit, death)
```

Comment:
the main point of the vignette is to calculate `time`, that's why I didn't select it.